### ISSUE TITLE
Getting "Error: EOF: end of file, read" on Windows

### ISSUE THREAD

**Issue by ot-jim-elphick** [aws/aws-cdk#11314]
### description of the bug:
returns JSII error on empty stack.  May be related to https://github.com/aws/aws-cdk/issues/5187.

stack trace references SyncStdio.readLine (myfolder\jsii\_embedded\jsii\jsii-runtime.js:13278:33)
Comment in code at 'throw e':
```
// HACK: node may set O_NONBLOCK on it's STDIN depending on what kind of input it is made
// of (see https://github.com/nodejs/help/issues/2663).
. . .
// "tracked" at https://github.com/aws/aws-cdk/issues/5187
```

### Reproduction Steps

(using powershell)  This follows example from https://docs.aws.amazon.com/cdk/latest/guide/ecs_example.html
```powershell
mkdir myfolder
cd myfolder
cdk init --language python
.env\Scripts\activate.ps1
python -m pip install -r requirements.txt
cdk synth
```
### What did you expect to happen?

return an empty stack with no error

### What actually happened?

```console
(.env) p$[myfolder]> cdk synth                                                                                                                      D:\d:\myfolder\.env\lib\site-packages\jsii\_embedded\jsii\jsii-runtime.js:13295
                    throw e;
                    ^

Error: EOF: end of file, read
    at Object.readSync (fs.js:592:3)
    at SyncStdio.readLine (D:\d:\myfolder\.env\lib\site-packages\jsii\_embedded\jsii\jsii-runtime.js:13278:33)
    at InputOutput.read (D:\d:\myfolder\.env\lib\site-packages\jsii\_embedded\jsii\jsii-runtime.js:13203:34)
    at KernelHost.run (D:\d:\myfolder\.env\lib\site-packages\jsii\_embedded\jsii\jsii-runtime.js:13021:32)
    at Immediate.<anonymous> (D:\d:\myfolder\.env\lib\site-packages\jsii\_embedded\jsii\jsii-runtime.js:13029:37)
    at processImmediate (internal/timers.js:461:21) {
  errno: -4095,
  syscall: 'read',
  code: 'EOF'
}
Resources:
  CDKMetadata:
    Type: AWS::CDK::Metadata
    Properties:
      Modules: aws-cdk=1.69.0,@aws-cdk/cloud-assembly-schema=1.69.0,@aws-cdk/core=1.69.0,@aws-cdk/cx-api=1.69.0,@aws-cdk/region-info=1.69.0,jsii-runtime=Python/3.7.4
    Metadata:
      aws:cdk:path: myfolder/CDKMetadata/Default
    Condition: CDKMetadataAvailable
```

### Environment

  - **CLI Version      :** aws-cli/2.0.61 Python/3.7.7 Windows/10 exe/AMD64
  - **Framework Version:** 1.69.0 (build 2b474b9)
  - **Node.js Version:** v14.15.0 
  - **OS               :** Windows 10
  - **Language (Version):** python 3.7.4 

### Other

Error is persistent and easily reproduced on my machine.
I re-installed/upgraded aws cli to 2.0.61
I re-installed node.js
I re-installed cdk
Note:  As noted in the section above, this 'hello world' level error still creates a stack.  However, when more code is added, more jsii errors occur and the stack fails.  Also, code that errors on my machine runs normally on other machines.

**Comment by RomainMuller** [comment]
I am unable to reproduce this problem on macOS/UNIX. Have to prepare a windows environment to attempt reproduction there; which might take a little while.

**Comment by RomainMuller** [comment]
Good news - I was able to reproduce it now (on Windows)! I'll try to figure out what's happening here...

**Comment by RomainMuller** [comment]
Alright - so the `EOF` error is thrown by `fs.readSync` on Windows when EOF. This does not appear to happen on *nixes and I'm also unable to directly reproduce this behavior (when I try to, the `fs.readSync` call properly returns `0` when EOF is reached, as documented).

This isn't quite the first oddity we have faced that *mostly* affects Windows users (as you've noted the hack we already have in place for `EAGAIN`)... I'm going to try a little harder to identify a reproduction and hopefully be able to file a bug report to `node`...

Interestingly, this error happens as the app has *correctly* finished (the Python process then closes the `node` process' `STDIN` to signal it's done and the `node` process should clean up after itself and exit - and this is what triggers the crash). This actually you'd be able to "work around" this by running your `app.py` and then running cdk with `-a cdk.out` until we have found a way to conclusively fix this problem.

If we're not finding a better solution fast enough, we'll have to actually identify this `EOF` error and actually handle it was if `fs.readSync` had returned 0....

**Comment by RomainMuller** [comment]
Alright I have been able to come up with a simple repro, which actually demonstrates the reason I could not "easily" reproduce previously is because of the difference in nature of the `process.stdin` when `node` is started directly as opposed to when Python creates a subprocess with piped in- and output.

**Comment by RomainMuller** [comment]
Filed a bug to NodeJS (nodejs/node#35997).

**Comment by ot-jim-elphick** [comment]
Merci!  Thank you for your help!  

This tells me to focus on node.js.  I will try using some previous version of node.js as a remedy.

**Comment by RomainMuller** [comment]
I *seem* unable to reproduce this issue on GitHub Actions where I'd run node `14.14.0`. I'm uncertain whether this is due to peculiarities of the GitHub Actions environment, or if node `14.14.0` is exempt from the faulty behavior.

**Comment by ot-jim-elphick** [comment]
@RomainMuller  Workaround: replace jsii 1.14.x distro in site-packages with 1.12 or 1.13 distro.  After swap, errors stop. 

TL;DR
I was able to swap out the 1.14.1 distro in the site-packages for the jsii 1.12 distro.   Same code that consistently threw the error reported above, now runs with no errors.  This behavior corresponds to the the reports from other devs (where I got the jsii 1.12 distro)  that they had no issues with the code.

Rolling back node.js or cdk to earlier versions had no effect because pip install - r requirements.txt always installed the 1.14.x distro.     

Longer term fix:  move dev platform from Windows to linux.

**Comment by kenchoong** [comment]
still see this in cdk version 1.73.0

**Comment by enrico-usai** [comment]
Same issue Python 3.9.0 and CDK 1.74.0 (build e86602f)

```
(.venv) C:\hello-cdk>cdk ls
C:\hello-cdk\.venv\lib\site-packages\jsii\_embedded\jsii\jsii-runtime.js:13295
                    throw e;
                    ^

Error: EOF: end of file, read
    at Object.readSync (fs.js:592:3)
    at SyncStdio.readLine (C:\hello-cdk\.venv\lib\site-packages\jsii\_embedded\jsii\jsii-runtime.js:13278:33)
    at InputOutput.read (C:\hello-cdk\.venv\lib\site-packages\jsii\_embedded\jsii\jsii-runtime.js:13203:34)
    at KernelHost.run (C:\hello-cdk\.venv\lib\site-packages\jsii\_embedded\jsii\jsii-runtime.js:13021:32)
    at Immediate.<anonymous> (C:\hello-cdk\.venv\lib\site-packages\jsii\_embedded\jsii\jsii-runtime.js:13029:37)
    at processImmediate (internal/timers.js:461:21) {
  errno: -4095,
  syscall: 'read',
  code: 'EOF'
}
hello-cdk

(.venv) C:\hello-cdk>python --version
Python 3.9.0

(.venv) C:\hello-cdk>cdk --version
1.74.0 (build e86602f)
```

**Comment by khomesh696** [comment]
> @RomainMuller Workaround: replace jsii 1.14.x distro in site-packages with 1.12 or 1.13 distro. After swap, errors stop.
> 
> TL;DR
> I was able to swap out the 1.14.1 distro in the site-packages for the jsii 1.12 distro. Same code that consistently threw the error reported above, now runs with no errors. This behavior corresponds to the the reports from other devs (where I got the jsii 1.12 distro) that they had no issues with the code.
> 
> Rolling back node.js or cdk to earlier versions had no effect because pip install - r requirements.txt always installed the 1.14.x distro.
> 
> Longer term fix: move dev platform from Windows to linux.

how should i install this jsii1.12 version in windows

**Comment by nulli2nd** [comment]
RE: "how should I install this jsii1.12 version in windows"

The 2nd answer in the stackoverflow question below expands on the 2 workarounds given.  

https://stackoverflow.com/questions/64694569/

**Comment by enrico-usai** [comment]
@nulli2nd should be enough to execute: ` pip install jsii==1.13` in your virtual environment.

**Comment by RomainMuller** [comment]
@ot-jim-elphick thanks for the deep dive, and sorry for not having gotten back there earlier... I'm going to investigate this further to try and understand what changed between `1.13.0` and `1.14.0` that can explain this behavior change.

**Comment by RomainMuller** [comment]
Is this still happening with `1.15.0`?

**Comment by enrico-usai** [comment]
I confirm that the issue is solved. Thanks!
```
$ cdk --version
1.74.0 (build e86602f)

$ pip list | grep jsii
jsii                               1.15.0

$ cdk ls
reading
hello-cdk
```
